benchmarks/distributed/rpc/ps/trainers/TrainerBase.py

```diff
@@ -0,0 +1,92 @@
+import functools
+from abc import ABC, abstractmethod
+
+from metrics.MetricsLogger import MetricsLogger
+
+
+class TrainerBase(ABC):
+
+    BATCH_LEVEL_METRIC = "batch_level_metric"
+    BATCH_ALL = "batch_all"
+    FORWARD_METRIC = "foward_metric"
+    FORWARD_PASS = "forward_pass"
+    BACKWARD_METRIC = "backward_metric"
+    BACKWARD = "backward"
+
+    def __init__(self, rank, metric_class="cuda", overwrite_metrics=False):
+        self.__metrics_logger = MetricsLogger(rank, metric_class, overwrite_metrics)
+
+    @abstractmethod
+    def train(self):
+        return
+
+    def record_start(self, metric_type, key, metric_name):
+        self.__metrics_logger.add_metric(
```

Comment:
curious, is there any reason that need a different name. The call trace is like:

record_start (TrainerBase) -> add_metric (MetricLogger) -> record_start(Metric)

Will there be any issue if we all stay with the same name?

Reply:
no but adding the `cuda_record_start` and `cpu_record_start` the call trace will be

cuda_record_start (TrainerBase) -> cuda_record_start (MetricLogger) -> record_start(CUDAMetric)
cpu_record_start (TrainerBase) -> cpu_record_start (MetricLogger) -> record_start(CPUMetric)

Reply:
call trace is now record_start (TrainerBase) -> record_start (MetricLogger) -> record_start(Metric)